app/graphql/types/custom_field_definition_type.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module Types
+  class CustomFieldDefinitionType < Types::BaseObject
+    description 'A custom field definition'
+
+    field :field_key, ID, null: false
+    field :label, String, null: false
+    field :data_type, String, null: false # Enum string, integer, boolean
```

Comment:
Doesn't have to be for this PR, but is a GraphQL enum type useful for data_type?

Reply:
@barmintor Did you create a ticket for creating enums?

Reply:
I did HYACINTH-456

Reply:
great, resolving this convo